yt_dlp/extractor/vidio.py

```diff
@@ -46,10 +49,44 @@ class VidioIE(InfoExtractor):
         'url': 'https://www.vidio.com/watch/1550718-stand-by-me-doraemon',
         'only_matching': True
     }]
+    _LOGIN_URL = 'https://www.vidio.com/users/login'
+    _NETRC_MACHINE = 'vidio'
+
+    def _login(self):
+        username, password = self._get_login_info()
+        if username is None:
+            return
+
+        login_page = self._download_webpage(
+            self._LOGIN_URL, None, 'Downloading log in page')
+
+        def is_logged():
+            return bool(self._download_json(
+                'https://www.vidio.com/interactions.json', None, False).get('current_user'))
+
+        # already logged in
+        if is_logged():
+            return
```

Comment:
```suggestion
        def is_logged_in():
            resp = self._download_json(
                'https://www.vidio.com/interactions.json', None, 'Checking if logged in', fatal=False) or {}
            return bool(resp.get('current_user'))

        if is_logged_in():
            return
```

Reply:
done, also pushed the check before the login page download